scalding-core/src/main/scala/com/twitter/scalding/macros/impl/TupleConverterImpl.scala

```diff
@@ -37,122 +37,185 @@ object TupleConverterImpl {
+          if (inOption) {
+            val cachedResult = newTermName(c.fresh(s"cacheVal"))
+            val boxed = box.map{ b => q"""$b($primitiveGetter)""" }.getOrElse(primitiveGetter)
+
+            val builder = q"""
           val $cachedResult: $boxedType = if(t.getObject($idx) == null) {
               null.asInstanceOf[$boxedType]
             } else {
               $boxed
             }
           """
-          (idx + 1,
-            Extractor(boxedType, q"$cachedResult"),
-            List(Builder(builder)))
-        } else {
-          (idx + 1, Extractor(outerTpe, primitiveGetter), List[Builder]())
+            (idx + 1,
+              Extractor(boxedType, q"$cachedResult"),
+              List(Builder(builder)))
+          } else {
+            (idx + 1, Extractor(outerTpe, primitiveGetter), List[Builder]())
+          }
         }
-      }
 
-      outerTpe match {
-        case tpe if tpe =:= typeOf[String] => getPrimitive(q"""t.getString(${idx})""", typeOf[java.lang.String], None)
-        case tpe if tpe =:= typeOf[Boolean] => getPrimitive(q"""t.getBoolean(${idx})""", typeOf[java.lang.Boolean], Some(q"_root_.java.lang.Boolean.valueOf"))
-        case tpe if tpe =:= typeOf[Short] => getPrimitive(q"""t.getShort(${idx})""", typeOf[java.lang.Short], Some(q"_root_.java.lang.Short.valueOf"))
-        case tpe if tpe =:= typeOf[Int] => getPrimitive(q"""t.getInteger(${idx})""", typeOf[java.lang.Integer], Some(q"_root_.java.lang.Integer.valueOf"))
-        case tpe if tpe =:= typeOf[Long] => getPrimitive(q"""t.getLong(${idx})""", typeOf[java.lang.Long], Some(q"_root_.java.lang.Long.valueOf"))
-        case tpe if tpe =:= typeOf[Float] => getPrimitive(q"""t.getFloat(${idx})""", typeOf[java.lang.Float], Some(q"_root_.java.lang.Float.valueOf"))
-        case tpe if tpe =:= typeOf[Double] => getPrimitive(q"""t.getDouble(${idx})""", typeOf[java.lang.Double], Some(q"_root_.java.lang.Double.valueOf"))
-        case tpe if tpe.erasure =:= typeOf[Option[Any]] && inOption =>
-          c.abort(c.enclosingPosition, s"Nested options do not make sense being mapped onto a tuple fields in cascading.")
-
-        case tpe if tpe.erasure =:= typeOf[Option[Any]] =>
-          val innerType = tpe.asInstanceOf[TypeRefApi].args.head
-
-          val (newIdx, extractor, builders) = matchField(innerType, idx, true)
-
-          val cachedResult = newTermName(c.fresh(s"opti"))
-          val extractorTypeVal: Tree = if (extractor.tpe =:= innerType)
-            extractor.toTree
-          else
-            q"${innerType.typeSymbol.companionSymbol}.unbox($extractor)"
-
-          val build = Builder(q"""
+        outerTpe match {
+          case tpe if tpe =:= typeOf[String] => getPrimitive(q"""t.getString(${idx})""", typeOf[java.lang.String], None)
+          case tpe if tpe =:= typeOf[Boolean] => getPrimitive(q"""t.getBoolean(${idx})""", typeOf[java.lang.Boolean], Some(q"_root_.java.lang.Boolean.valueOf"))
```

Comment:
we could support `Char`, which `t.getChar` should work. For whatever reason, `Byte` is not supported by cascading.